packages/terra-button/src/Button.jsx

```diff
@@ -115,221 +115,194 @@ const defaultProps = {
   variant: ButtonVariants.NEUTRAL,
 };
 
-class Button extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { active: false, focused: false };
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-    this.handleKeyUp = this.handleKeyUp.bind(this);
-    this.handleOnBlur = this.handleOnBlur.bind(this);
-    this.handleFocus = this.handleFocus.bind(this);
-    this.handleMouseDown = this.handleMouseDown.bind(this);
-    this.handleClick = this.handleClick.bind(this);
-
-    this.shouldShowFocus = true;
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (prevState.focused && nextProps.isDisabled) {
-      return { focused: false };
+const Button = ({
+  icon,
+  isBlock,
+  isCompact,
+  isDisabled,
+  isIconOnly,
+  isReversed,
+  text,
+  type,
+  variant,
+  href,
+  onClick,
+  onMouseDown,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  refCallback,
+  title,
+  ...customProps
+}) => {
+  const theme = React.useContext(ThemeContext);
+  const [focused, setFocused] = useState(false);
+  const [active, setActive] = useState(false);
+
+  useEffect(() => {
+    setActive(false);
+    if (isDisabled) {
+      setFocused(false);
     }
-    return null;
-  }
+  }, [isDisabled]);
 
-  handleOnBlur(event) {
-    this.setState({ focused: false });
+  const handleOnBlur = (event) => {
+    setFocused(false);
 
-    if (this.props.onBlur) {
-      this.props.onBlur(event);
+    if (onBlur) {
+      onBlur(event);
     }
-  }
+  };
 
-  handleClick(event) {
+  const handleClick = (event) => {
     // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
     // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
     // This will put focus on the button when clicked if it is not currently the active element.
     if (document.activeElement !== event.currentTarget) {
-      this.shouldShowFocus = false;
       event.currentTarget.focus();
-      this.shouldShowFocus = true;
     }
 
-    if (this.props.onClick) {
-      this.props.onClick(event);
+    if (onClick) {
+      onClick(event);
     }
-  }
+  };
 
-  handleKeyDown(event) {
+  const handleKeyDown = (event) => {
     // Add active state to FF browsers
     if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE) {
```

Comment:
I know that this is not your code, but we should not need the nativeEvent to access the keyCode.